src/test/java/com/heatheryou/httpserver/RouterTest.java

```diff
@@ -24,10 +28,46 @@ public void ItReturnsTrueIfRequestedMethodAndUriAreSupported() {
     }
 
     @Test
-    public void ItReturnsFalseIfRequestedMethodAndUriNotSupported() {
+    public void ItReturnsFalseIfRequestedUriIsValidButMethodNotSupported() {
         Router router = new Router();
         boolean actual = router.isValidRequest("/simple_get", "POST");
         assertFalse(actual);
     }
 
+    @Test
+    public void ItReturnsFalseIfRequestedUriAndMethodNotSupported() {
+        Router router = new Router();
+        boolean actual = router.isValidRequest("/invalid_uri", "POST");
+        assertFalse(actual);
+    }
+
+    @Test
+    public void ItReturnsTrueIfValidUriIsRequested() {
+        Router router = new Router();
+        boolean actual = router.isValidRoute("/simple_get");
+        assertTrue(actual);
+    }
+
+    @Test
+    public void ItReturnsFalseIfInvalidUriIsRequested() {
+        Router router = new Router();
+        boolean actual = router.isValidRoute("/invalid_uri");
+        assertFalse(actual);
+    }
+
+    @Test
+    public void ItReturnsMethodNotAllowedHandlerIfValidUriWithInvalidMethodIsRequested() {
+        Router router = new Router();
+        Request request = new Request("/simple_get", "POST");
+        IHandler actual = router.handleRequest(request);
+        assertThat(actual, instanceOf(MethodNotAllowedHandler.class));
+    }
+
+    @Test
+    public void ItReturnsNoRouteFoundHandlerIfInvalidUriWithInvalidMethodIsRequested() {
+        Router router = new Router();
+        Request request = new Request("/invalid_route", "POST");
+        IHandler actual = router.handleRequest(request);
+        assertThat(actual, instanceOf(NoRouteFoundHandler.class));
+    }
```

Comment:
The first two tests are testing the method `isValidRequest`,  the next two are testing `isValidRoute` and the last two are testing `handleRequest`. Naming the tests in such a way to reflect which methods are actually being tested, rather than using "it" for all of them, will make your tests easier to follow. If you're ambitious, you can even nest the tests so that the tests for each method are grouped together.

Reply:
Replaced all the "its" with the method being tested. Would love to learn more about creating nested tests when you have a chance 😊

Reply:
Fixed in c79b1ef4ae234d6f8a8f21934be0061355a03ffc

Reply:
JUnit syntax for nested tests: https://junit.org/junit5/docs/current/user-guide/#writing-tests-nested

You probably don't need to use it at this stage, but it's worth keeping in mind, since it might come in handy when your test files grow longer.